lib/src/components/containers/newMember.dart

```diff
@@ -461,6 +503,127 @@ class NewMember extends PComponent<NewMemberProps> {
         ]
     ];
 
+  //Input validation for each field 
+
+  //Validation for first name
+  void _firstNameValidation(_) {
+    //Gets first field
+    InputElement first = querySelector('#fName-input');
+    //Bool for setting state
+    bool isValid;
+    //Checks if value is blank
+    if(first.value == '') {
+      isValid = false;
+    } else isValid = true;
+    //Sets new state
+    setState((NewMemberProps, NewMemberState) => 
+      NewMemberState..firstNameIsValid = isValid);
+  }
+
+  //Validation for last name
+  void _lastNameValidation(_) {
+    //Gets last field
+    InputElement last = querySelector('#lName-input');
+    //Bool for setting state
+    bool isValid;
+    //Checks if value is blank
+    if(last.value == '') {
+      isValid = false;
+    } else isValid = true;
+    //Sets new state
+    setState((NewMemberProps, NewMemberState) => 
+      NewMemberState..lastNameIsValid = isValid);
+  }
+
+  //Validation for email using Spencer's function from constants.dart
+  void _emailValidator(_) {
+    //Gets email field
+    InputElement email = querySelector('#email-input');
+    //Bool for setting state
+    bool isValid;
+    if(emailIsValid(email.value)) {
+      isValid = true;
+    } else isValid = false;
+    setState((NewMemberProps, NewMemberState) =>
+      NewMemberState..emailIsValid = isValid);
+  }
+
+  //Validation for phone numbers
+  //Area code is required
+
+  //Possible formats
+  //4252734489
+  //14252734489
+  //(425)2734489
+  //1(425)2734489
+  //425-273-4489
+  //1-425-273-4489
+  //+1-425-273-4489
+  
+   //Validation for phone number, doesnt check if blank
+  void _phoneNumberValidator(_) {
+    //Gets phone field and then value from field
+    InputElement phone = querySelector('#phoneNum-input');
+    String value = phone.value;
+    //Bool for setting state
+    bool isValid;
+    //If blank exit
+    if(value == '') {
+      return;
+    }
+    //Acual validation
+    //Splits string into a list
+    List<String> temp = value.split('');
+    //Counts digits in input string
+    int count = 0;
+    for (String x in temp) {
+      if(int.tryParse(x) != null) {
+        count++;
+      }
+    }
+    if(count == 10 || count == 11) {
+      isValid = true;
+    } else {isValid = false;}
+    //Sets state
+    setState((NewMemberProps, NewMemberState) =>
+      NewMemberState..phoneNumberIsValid = isValid);
+  }
+
+  //Validation for cell number, doesnt check if blank
+  void _cellNumberValidator(_) {
+    //Gets cell field and then value from field
+    InputElement cell = querySelector('#cellNum-input');
+    String value = cell.value;
+    //Bool for setting state
+    bool isValid;
+    //Exits if blank
+    if(value == '') {
```

Comment:
Don't we still want to `setState` here, since if I enter `foo` into the box, then delete it, it will stay in an error state?